yt/frontends/boxlib/data_structures.py

```diff
@@ -1650,11 +1644,13 @@ def _parse_parameter_file(self):
                     self.parameters[l[0].strip()] = l[1].strip()
 
         # set the periodicity based on the integer BC runtime parameters
-        is_periodic = self.parameters["geometry.is_periodic"].split()
-        periodicity = [bool(val) for val in is_periodic]
-        for _ in range(self.dimensionality, 3):
-            periodicity += [True]  # pad to 3D
-        self.periodicity = ensure_tuple(periodicity)
+        self.periodicity = [True] * 3
```

Comment:
The default in AMReX is `False` if not specified: https://amrex-codes.github.io/amrex/docs_html/InputsProblemDefinition.html

Reply:
Thank you for checking this ! I assumed the dominant behavior was correct but I guess that's one more bug to take down here. Will fix as soon as I get my hands on my laptop.

Reply:
I went back and applied the same fix to BoxlibDataset itself as well as Castro following the docs.